test/profileCache.js

```diff
@@ -207,6 +207,56 @@ describe('profile cache', function() {
     });
   });
 
+  it('should invalidate cache when profileChangedAt is greater than cached version', function (done) {
+    const userid = uid();
+    this.timeout(5000);
+    Server.server.initialize(() => {
+      let lastModified;
+      // first req, store last modified header
+      mock.profileChangedAt(MOZILLA_EMAIL, Date.now());
+      makeProfileReq(userid)
+        .then(res => {
+          assert.ok(res.headers['last-modified']);
+          lastModified = res.headers['last-modified'];
+          return P.delay(1000);
+        })
+        .then(() => {
+          // second request verify cached result was returned
+          return makeProfileReq(userid);
+        })
+        .then(res => {
+          assert.ok(res.headers['last-modified']);
+          assert.equal(res.headers['last-modified'], lastModified);
+          mock.token({
+            user: userid,
+            scope: ['profile:avatar:write']
+          });
+          // upload avatar (should invaldate cache)
+          mock.image(imageData.length);
```

Comment:
I'm new to this repo too and am probably missing something obvious, this test looks the same as the previous test. Is the cache being invalidated because of the new code or because the avatar is uploaded?

Reply:
Heh, no it is the same test, didn't get a chance to update it yet. I was going to use previous test as a basis for this one.